### ISSUE TITLE
[bug] sendTransfer fails with user rejection error even if approved.

### ISSUE THREAD

**Issue by JCaraballo113** [secretkeylabs/sats-connect#167]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SATS-CONNECT Version

2.8.0

### Current Behavior

When I call the sendTransfer method and approve the transaction I suddenly get a "User rejected request to send transfer"

### Expected Behavior

When I call the sendTransfer method and approve the transaction my transaction should be successful

### Steps To Reproduce

Connect Xverse, send a transfer

**Comment by teebszet** [comment]
hi @JCaraballo113, thanks for the report!

could you give us a bit more information on:
- the version of Xverse wallet you're using
- the code where you're calling sendTransfer

additionally, what happens when you connect your wallet to this example app: https://sats-connect.netlify.app/bitcoin-methods and try to use the Send BTC UI?

fyi, the example app is built from this code here: https://github.com/secretkeylabs/sats-connect/blob/13e9eefff1ea141bd52fad689c99e6612e76c13c/example/src/components/bitcoin/SendBtc/index.tsx#L35-L53

thanks

**Comment by JCaraballo113** [comment]
@teebszet 

the version of Xverse wallet you're using -> 0.40.0
the code where you're calling sendTransfer -> Same as example.

I actually went to the netlify app and same issue. Attaching screenshots:

<img width="547" alt="Screenshot 2024-08-28 at 8 55 23 AM" src="https://github.com/user-attachments/assets/663f4b53-406f-4a85-b667-166884dbf3e3">
<img width="484" alt="Screenshot 2024-08-28 at 8 55 37 AM" src="https://github.com/user-attachments/assets/8e667e48-1efa-4129-a388-8abf0f6c468b">

**Comment by eustacia-vye** [comment]
I am experiencing the same behavior.

sats-connect: v2.8.0
XVerse Wallet v0.40.0

```
const response = await Wallet.request("sendTransfer", {
  recipients: [
    {
      address: recipient,
        amount: Number(amount),
    },
  ],
});
console.log(response);
```
Transaction in XVerse:
<img width="350" alt="Approve Transaction" src="https://github.com/user-attachments/assets/3f6085d8-48cc-4f8d-9411-3c50d2cafd1b">

Transaction Submitted (no response yet to `sendTransfer`:
<img width="350" alt="Transaction Submitted" src="https://github.com/user-attachments/assets/c5ca0c40-e197-498d-aebd-dc9b7b2f0106">


After clicking "Close", `sendTransfer` returns, error in console, not transaction hash:
<img width="501" alt="Console Log" src="https://github.com/user-attachments/assets/68b946cb-71c1-44d4-b224-24c23ca07fe8">

**Comment by JCaraballo113** [comment]
Any updates on this @teebszet ?

**Comment by walkerliu01** [comment]
I  have the same problem

**Comment by afxal** [comment]
Tested an old version of sats-connect and the recent version with Xverse app on mobile, works perfectly. 

On chromium browsers it fails to recognize the success events, I can confirm with the most recent chrome + Xverse plugin.

**Comment by 0xAlmir** [comment]
Having the same problem as well
Xverse Version 0.41
sats-connect: 2.8.0
on Chrome

**Comment by plasmatech8** [comment]
The app I was building just broke recently due to the close button unexpectedly causing user rejection error.

I haven't changed the version of the `sats-connect` library. Is this is a problem with the XVerse Wallet extension, rather than sats-conenct library?

**Comment by JCaraballo113** [comment]
@plasmatech8  based on my diagnosis, yes. if you call the window injected bitcoin provider by Xverse you will get a user rejection error even with a confirmation. So this is without using the sats-connect library.

**Comment by m-aboelenein** [comment]
Hi everyone, we've confirmed the issue with the Xverse Web Extension. Our team has identified the cause, and we're planning to include the fix in the upcoming release scheduled for this week. Thank you for bringing it to our attention! 🙏

**Comment by teebszet** [comment]
this should be patched from the Xverse Web Extension v0.42.0 onwards. thank you!
